components/dashboard/src/provider-utils.tsx

```diff
@@ -26,10 +28,71 @@ function simplifyProviderName(host: string) {
         case "gitlab.com":
             return "GitLab"
         case "bitbucket.org":
-            return "BitBucket"
+            return "Bitbucket"
         default:
             return host;
     }
 }
 
-export { iconForAuthProvider, simplifyProviderName }
\ No newline at end of file
+async function openAuthorizeWindow({ host, scopes, onSuccess, onError }: { host: string, scopes?: string[], onSuccess?: () => void, onError?: (error?: string) => void }) {
+    const returnTo = gitpodHostUrl.with({ pathname: 'login-success' }).toString();
+    const url = gitpodHostUrl.withApi({
+        pathname: '/authorize',
+        search: `returnTo=${encodeURIComponent(returnTo)}&host=${host}&override=true&scopes=${(scopes || []).join(',')}`
+    }).toString();
+
+    const newWindow = window.open(url, "gitpod-connect");
+    if (!newWindow) {
+        console.log(`Failed to open the authorize window for ${host}`);
+        onError && onError("failed");
+        return;
+    }
+
+    const eventListener = (event: MessageEvent) => {
+        // todo: check event.origin
+
+        if (event.data === "auth-success") {
+            window.removeEventListener("message", eventListener);
+
+            if (event.source && "close" in event.source && event.source.close) {
+                console.log(`Authorization OK. Closing child window.`);
+                event.source.close();
+            } else {
+                // todo: add a button to the /login-success page to close, if this should not work as expected
+            }
+            onSuccess && onSuccess();
+        }
+    };
+    window.addEventListener("message", eventListener);
+
+    /**
+     * As a fallback solution to the cross window messaging, we can try polling for the expected location.
+     */
```

Comment:
Please comment why do we need a fallback solution? What are the reasons the window messaging could possibly not work ok?

Reply:
removing, as I haven't seen that during testing at all.